edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/DeviceScopeAuthenticator.cs

```diff
@@ -102,25 +103,50 @@ async Task<(bool isAuthenticated, bool shouldFallback)> AuthenticateInternalAsyn
             }
         }
 
-        async Task<(bool isAuthenticated, bool serviceIdentityFound)> AuthenticateWithServiceIdentity(T credentials, string serviceIdentityId, bool syncServiceIdentity)
+        async Task<(bool isAuthenticated, bool serviceIdentityFound)> AuthenticateWithAuthChain(T credentials, string serviceIdentityId, bool syncServiceIdentity)
         {
-            Option<string> authChain = await this.deviceScopeIdentitiesCache.GetAuthChain(serviceIdentityId);
+            // Try authenticate against the originating identity first
+            (bool isAuthenticated, bool serviceIdentityFound) = await this.AuthenticateWithServiceIdentity(credentials, serviceIdentityId, syncServiceIdentity, false);
+
+            if (!this.nestedEdgeEnabled)
+            {
+                // Legacy path
+                return (isAuthenticated, serviceIdentityFound);
+            }
 
-            if (this.nestedEdgeEnabled && !authChain.HasValue)
+            // For nested Edge, we need to check that an authchain exists
+            Option<string> authChain = await this.deviceScopeIdentitiesCache.GetAuthChain(serviceIdentityId);
+            if (!authChain.HasValue)
             {
                 // The identity is not within our nested hierarchy
                 return (false, false);
             }
 
+            var authChainIds = authChain.OrDefault().Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries).ToList();
+            if (!isAuthenticated && authChainIds.Count() > 2)

```

Comment:
we should be getting the actor from the token's audience, then verifying that the actor is in the authchain.

Reply:
I don't understand "authChainIds.Count() > 2"? why 2?

Reply:
because there should always be at least 3 ids in the authchain for OnBehalfOf, [original]/[actor]/[parent].

But this doesn't apply anymore, I'm now just getting the actor out of the credential.

---
In reply to: [451892190](https://github.com/Azure/iotedge/pull/3190#discussion_r451892190) [](ancestors = 451892190)